src/Tribe/Attendee_Registration/Shortcode.php

```diff
@@ -14,8 +14,42 @@ public function hook() {
 		}
 
 		add_shortcode( $this->shortcode_name, [ $this, 'render' ] );
+
+		add_action( 'wp_enqueue_scripts', [ $this, 'maybe_enqueue_scripts'] );
+	}
+
+	public function maybe_enqueue_scripts() {
+		if (
+			is_archive()
+			|| is_admin()
+		) {
+			return;
+		}
+
+		$shortcode_page = tribe_get_option( 'ticket-attendee-page-id', false );
+		// option not set
+		if ( ! $shortcode_page ) {
+			return;
+		}
+
+		$slug = parse_url( $_SERVER[ 'REQUEST_URI' ], PHP_URL_PATH );
+		// not on a page we can parse
+		if ( ! $slug ) {
+			return;
+		}
+
+		$page = get_page_by_path( $slug, OBJECT );
```

Comment:
Shouldn't this be available somewhere already on the current page, if it's a page? `get_queried_object_id()`?

Reply:
you are correct sir... 🤦‍♂